annotations/views/rest_views.py

```diff
@@ -36,6 +36,55 @@
 logger.setLevel(settings.LOGLEVEL)
 
 
+# Custom permission class to check if user is collaborator or owner
+class IsCollaboratorOrOwner(IsAuthenticatedOrReadOnly):
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):
+            return False
+            
+        # Allow GET requests for authenticated users
+        if request.method in ['GET', 'HEAD', 'OPTIONS']:
+            return True
```

Comment:
I'm confused, why would it matter if this is a GET or POST request?

Reply:
Hm, ok, I think I slowly understand, I think it's a naming issue. The class name suggests this will just check if the user is collaborator and owner, not that it checks specific texts for edit permissions (which I think this is doing?). Either it should check in general if the user can see/edit the text or the class name needs to be way more specific. In either case, the class name needs to be changed though.